src/service/link-rewriter/link-rewriter-manager.js

```diff
@@ -0,0 +1,231 @@
+   * defined priority order (in link attribute or global meta tag).
+   * If a LinkRewriter decides to not replace the url,  we will ask the next
+   * one if it wants to replace the link until one replacement is made.
+   *
+   * A "CLICK" event will also be dispatched to notify listeners that a click
+   * on an anchor has happened. This should mostly be used to send click
+   * tracking requests, handlers of this events should not
+   * mutate the anchor!
+   * @param {HTMLElement} anchor
+   */
+  maybeRewriteLink(anchor) {
+    const suitableLinkRewriters = this.getSuitableLinkRewritersForLink_(anchor);
+    if (suitableLinkRewriters.length) {
+      let chosenLinkRewriter = null;
+
+      // Iterate by order of priority until one of the linkRewriter
+      // replaces the link.
+      for (let i = 0; i < suitableLinkRewriters.length; i++) {
+        const hasReplaced = suitableLinkRewriters[i].rewriteAnchorUrl(anchor);
+        if (hasReplaced) {
+          chosenLinkRewriter = suitableLinkRewriters[i];
+          break;
+        }
+      }
+      const linkRewriterId = chosenLinkRewriter ? chosenLinkRewriter.id : null;
+
+      // Emit click event for analytics purposes only,
+      // anchor should not me mutated by handlers.
+      const eventData = {
+        linkRewriterId,
+        anchor,
+      };
+
+      suitableLinkRewriters.forEach(linkRewriter => {
+        linkRewriter.events.send(EVENTS.CLICK, eventData);
+      });
+    }
+  }
+
+  /**
+   * @private
+   * Extract the priority list from the optional html meta tag.
+   * The meta tag should contain a whitespace separated list of
+   * LinkRewrited.id.
+   * @param {!../ampdoc-impl.AmpDoc} ampdoc
+   * @return {Array<string>}
+   */
+  getPriorityList_(ampdoc) {
+    const docInfo = Services.documentInfoForDoc(ampdoc);
+    const value = docInfo.metaTags[PRIORITY_META_TAG_NAME];
+
+    return value ? value.trim().split(/\s+/) : [];
+  }
+
+  /**
+   * @private
+   * Listen for DOM_UPDATE event.
+   * @param {*} iframeDoc
+   */
+  installGlobalEventListener_(iframeDoc) {
+    iframeDoc.addEventListener(AmpEvents.DOM_UPDATE,
+        this.onDomChanged_.bind(this));
+  }
+
+  /**
+   * @private
+   * Notify all the registered LinkRewriter when new elements have been added
+   * to the page so they can re-scan the page to find potential new links to
+   * replace.
+   */
+  onDomChanged_() {
+    this.linkRewriters_.forEach(linkRewriter => {
+      linkRewriter.onDomUpdated();
+    });
+  }
+
+
+  /**
+   * @private
+   * Extract the optional priority list for this specific anchor
+   * from its attribute. The 'data-link-rewriters' attribute should
+   * contain a whitespace separated list of LinkRewrited.id.
+   * @param {HTMLElement} anchor
+   * @return {Array<string>}
+   */
+  parseLinkRewriterPriorityForAnchor_(anchor) {
+    const dataValue = anchor.getAttribute('data-link-rewriters');
+    if (!anchor || !dataValue) {
```

Comment:
For extra safety in case of someone leaves a value such as `data-link-rewriters=" "`:

```js
const dataValue = anchor.hasAttribute('data-link-rewriters')
  ? anchor.getAttribute('data-link-rewriters').trim()
  : null;

if (...) { ... }

return dataValue.split(/\s+/);
```